crates/iceberg/src/spec/schema.rs

```diff
@@ -84,28 +206,853 @@ impl Schema {
     pub fn schema_id(&self) -> i32 {
         self.schema_id
     }
+
+    /// Returns [`r#struct`].
+    #[inline]
+    pub fn as_struct(&self) -> &StructType {
+        &self.r#struct
+    }
+
+    /// Get field id by full name.
+    pub fn field_id_by_name(&self, name: &str) -> Option<i32> {
+        self.name_to_id.get(name).cloned()
```

Comment:
It's a rust specific problem, `Hashmap::get(key)` returns a reference to value, e.g. `&i32` here. We use `cloned` to avoid returning a reference to `i32`, it's not a big problem since it's just a dereference to a pointer.

Reply:
`self.name_to_id.get()` will return an `Option<&i32>` here so it's better to perform a `clone` here to make sure users can use `i32` directly instead of `*v` by hand. It's a zero cost operation. It's already immutable now (users can't change the value in our `name_to_id`).

> I'm guessing we can use `copied()` here (alought it doesn't have real meanings).

Reply:
Yeah, `copied()` is better. Why clippy doesn't warn this 😁

Reply:
Thanks for the explanation, learning every day!